### PR TITLE
Update Android build environment

### PR DESCRIPTION
**Pull request by Gymnasiast**
Contains more backported stuff from #16107. Should also make it possible to enable std::filesystem for Android (see https://github.com/OpenRCT2/OpenRCT2/pull/16415).

Do not squash merge.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
The resulting app now starts up correctly on my test device (Nexus 6). Earlier I ran into problems due to missing the arm7 architecture and falling foul of new storage rules that Android 11 introduced.

The compilation does appear to be a little on the slow side, though. I’m going to take a look and try to find out why.

**Review by tupaschoal — COMMENTED** [review]
There is one conflict due to #16770 being merged, but I think it's simple to solve, can you look into it?

**Comment by Gymnasiast** [comment]
It should all be fine, but I’d like to test the artifact before this is merged, so could you approve if it’s all OK to you? I will then merge it after the test.

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**
